src/main/scala/junctions/util.scala

```diff
@@ -244,128 +244,3 @@ class DecoupledHelper(val rvs: Seq[Bool]) {
     (rvs.filter(_ ne exclude) ++ includes).reduce(_ && _)
   }
 }
-
-class MultiWidthFifo(inW: Int, outW: Int, n: Int) extends Module {
```

Comment:
Since you've already created a separate top-level util package, we might want to move everything from this file into that package. There's nothing here that's specific to junctions. Some of the stuff from uncore.util could be moved as well.